src/main/java/htsjdk/samtools/TextTagCodec.java

```diff
@@ -149,12 +150,12 @@ public String encodeUntypedTag(final String tagName, final Object value) {
      */
     public Map.Entry<String, Object> decode(final String tag) {
         final int numFields = StringUtil.splitConcatenateExcessTokens(tag, fields, ':');
-        if (numFields != TextTagCodec.NUM_TAG_FIELDS) {
+        if (numFields != TextTagCodec.NUM_TAG_FIELDS && numFields != TextTagCodec.NUM_TAG_FIELDS - 1) {
```

Comment:
Not a big deal, but might be simpler to test
if (numFields < TextTagCodec.NUM_TAG_FIELDS - 1)